docs/book/stacks-and-components/component-guide/orchestrators/skypilot-vm.md

```diff
@@ -371,6 +371,46 @@ skypilot_settings = SkypilotAzureOrchestratorSettings(
 {% endtab %}
 {% endtabs %}
 
+
+One of the key features of the SkyPilot VM Orchestrator is the ability to run each step of a pipeline on a separate VM with its own specific settings. This allows for fine-grained control over the resources allocated to each step, ensuring that each part of your pipeline has the necessary compute power while optimizing for cost and efficiency.
+
+## Configuring Step-Specific Resources
+
+The SkyPilot VM Orchestrator allows you to configure resources for each step individually. This means you can specify different VM types, CPU and memory requirements, and even use spot instances for certain steps while using on-demand instances for others.
+
+To configure step-specific resources, you can pass a `SkypilotBaseOrchestratorSettings` object to the `settings` parameter of the `@step` decorator. This object allows you to define various attributes such as `instance_type`, `cpus`, `memory`, `use_spot`, `region`, and more.
+
+Here's an example of how to configure specific resources for a step for the AWS cloud:
+
+```python
+from zenml.integrations.skypilot.flavors.skypilot_orchestrator_aws_vm_flavor import SkypilotAWSOrchestratorSettings
+
+# Settings for a specific step that requires more resources
+high_resource_settings = SkypilotAWSOrchestratorSettings(
+    instance_type='t2.2xlarge',
+    cpus=8,
```

Comment:
shouldnt we take this from `ResourceSettings` actually now that I think about it?

Reply:
This and the GPU

Reply:
I think it would be better to distinguish between both of them, because `ResourceSettings` can be defined for things like pods which would allow random resource given, while for Skypilot most of these parameters must match if supplied together

Reply:
IMO the `cpu_count` and `memory` in ResourceSettings should be mapped here... what is the point of ResourceSettings otherwise? @schustmi would love your feedback